bundles/org.eclipse.equinox.launcher/src/org/eclipse/equinox/launcher/KeyStoreUtil.java

```diff
@@ -0,0 +1,168 @@
+
+		if (Boolean.FALSE.toString().equals(System.getProperty(PROP_LOAD_OS_TRUST_STORE_BY_DEFAULT, null))) {
+			return;
+		}
+
+		List<KeyStoreAndPassword> keyStores = new ArrayList<>();
+		keyStores.add(new KeyStoreAndPassword(null, null)); // null will loads JVM cacerts OR store indicated by "javax.net.ssl.trustStore" properties
+		if (System.getProperty("javax.net.ssl.trustStore", "").isEmpty()) { //NOPMD
+			if (Constants.OS_MACOSX.equals(os)) {
+				keyStores.add(createKeyStore("KeychainStore", "Apple"));
+			} else if (Constants.OS_WIN32.equals(os)) {
+				keyStores.add(createKeyStore("Windows-ROOT", null));
+			}
+		}
+		List<X509TrustManager> trustManagers = new ArrayList<>();
+		for (KeyStoreAndPassword storeAndPassword : keyStores) {
+			trustManagers.add(createX509TrustManager(storeAndPassword.keyStore()));
+		}
+		TrustManager[] tm = {new CollectionTrustManager(trustManagers)};
+
+		KeyManager[] km = {};
+		KeyStoreAndPassword keyStore = createKeyStoreFromSystemProperties();
+		if (keyStore != null) {
+			km = new KeyManager[] {createX509KeyManager(keyStore.keyStore(), keyStore.password())};
+		}
+
+		SSLContext sslContext = SSLContext.getInstance("TLS");
+		initSSLContext(sslContext, tm, km, null);
+		SSLContext.setDefault(sslContext);
+	}
+
+	private KeyStoreAndPassword createKeyStore(String type, String provider) throws GeneralSecurityException, IOException {
+		KeyStore keyStore;
+		if (provider == null) {
+			keyStore = KeyStore.getInstance(type);
+		} else {
+			keyStore = KeyStore.getInstance(type, provider);
+		}
+		keyStore.load(null, null);
+		return new KeyStoreAndPassword(keyStore, null);
+	}
+
+	/*package, for test*/ X509TrustManager createX509TrustManager(KeyStore keyStore) throws GeneralSecurityException {
+		TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+		tmf.init(keyStore);
+		return Arrays.stream(tmf.getTrustManagers()).filter(X509TrustManager.class::isInstance) //
+				.map(X509TrustManager.class::cast) //
+				.findFirst().orElse(null);
+	}
+
+	/*package, for test*/ X509KeyManager createX509KeyManager(KeyStore keyStore, char[] password) throws GeneralSecurityException {
+		KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+		kmf.init(keyStore, password);
+		return Arrays.stream(kmf.getKeyManagers()).filter(X509KeyManager.class::isInstance) //
+				.map(X509KeyManager.class::cast) //
+				.findFirst().orElse(null);
+	}
+
+	/**
+	 * Coding based on
+	 * sun.security.ssl.SSLContextImpl.DefaultManagersHolder.getKeyManagers()
+	 * with minor adjustments (access properties directy without
+	 * AccessController, return nothing if properties not set).
+	 */
+	@SuppressWarnings("nls")
+	private KeyStoreAndPassword createKeyStoreFromSystemProperties() throws GeneralSecurityException, IOException { //NOPMD
+		String p11KeyStore = "PKCS11";
+		String none = "NONE";
```

Comment:
:thinking: These are effectively constants, better to make them `final`, at least, or convert to real constants?